packages/mdc-textfield/adapter.js

```diff
@@ -0,0 +1,194 @@
+   */
+  deregisterTextFieldInteractionHandler(type, handler) {}
+
+  /**
+   * Emits a custom event "MDCTextfield:icon" denoting a user has clicked the icon.
+   */
+  notifyIconAction() {}
+
+  /**
+   * Adds a class to the bottom line element.
+   * @param {string} className
+   */
+  addClassToBottomLine(className) {}
+
+  /**
+   * Removes a class from the bottom line element.
+   * @param {string} className
+   */
+  removeClassFromBottomLine(className) {}
+
+  /**
+   * Adds a class to the help text element. Note that in our code we check for
+   * whether or not we have a help text element and if we don't, we simply
+   * return.
+   * @param {string} className
+   */
+  addClassToHelptext(className) {}
+
+  /**
+   * Removes a class from the help text element.
+   * @param {string} className
+   */
+  removeClassFromHelptext(className) {}
+
+  /**
+   * Returns whether or not the help text element contains the given class.
+   * @param {string} className
+   * @return {boolean}
+   */
+  helptextHasClass(className) {}
+
+  /**
+   * Registers an event listener on the native input element for a given event.
+   * @param {string} evtType
+   * @param {function(!Event)} handler
+   */
+  registerInputInteractionHandler(evtType, handler) {}
+
+  /**
+   * Deregisters an event listener on the native input element for a given event.
+   * @param {string} evtType
+   * @param {function(!Event)} handler
+   */
+  deregisterInputInteractionHandler(evtType, handler) {}
+
+  /**
+   * Registers an event listener on the bottom line element for a "transitionend" event.
+   * @param {function(!Event)} handler
+   */
+  registerTransitionEndHandler(handler) {}
+
+  /**
+   * Deregisters an event listener on the bottom line element for a "transitionend" event.
+   * @param {function(!Event)} handler
+   */
+  deregisterTransitionEndHandler(handler) {}
+
+  /**
+   * Sets an attribute with a given value on the bottom line element.
+   * @param {string} attr
+   * @param {string} value
+   */
+  setBottomLineAttr(attr, value) {}
+
+  /**
+   * Sets an attribute with a given value on the help text element.
+   * @param {string} name
+   * @param {string} value
+   */
+  setHelptextAttr(name, value) {}
+
+  /**
+   * Removes an attribute from the help text element.
+   * @param {string} name
+   */
+  removeHelptextAttr(name) {}
+
+  /**
+   * Returns an object representing the native text input element, with a
+   * similar API shape. The object returned should include the value, disabled
+   * and badInput properties, as well as the checkValidity() function. We never
+   * alter the value within our code, however we do update the disabled
+   * property, so if you choose to duck-type the return value for this method
+   * in your implementation it's important to keep this in mind. Also note that
+   * this method can return null, which the foundation will handle gracefully.
+   * @return {HTMLInputElement|NativeInputType}
```

Comment:
Add [explicit nullable operator(s)](https://google.github.io/styleguide/jsguide.html#jsdoc-nullability):

`@return {?HTMLInputElement|?NativeInputType}`
or
`@return {!HTMLInputElement|!NativeInputType|null}`
or
`@return {HTMLInputElement|NativeInputType|null}`

(I'm not sure which variation is more readable, correct, and Closure-friendly...)

Reply:
Done. I think `null` is generally discouraged when `?` would suffice, so I went with `{?A|?B}`.